shashi/practiceTest2.js

```diff
@@ -0,0 +1,67 @@
+//calculate Factorial of a number
+const calcFactorial = num => {
+  if(num < 0){
+    return `Not a value number to calculate factorial`;
+  } 
+  if (num === 0 || num === 1){
+    return 1;
+  } 
+  if(num > 1) {
+   return num * calcFactorial(num-1);
+  }
+}
+
+console.log(calcFactorial(5));
+
+//calculate power of a number
+const power = (num, times) => {
+  if(times < 0 || num < 0){
+    return -1;
+  }
+  if ( num === 0 || times === 0){
+    return 0;
+  } 
+  if ( num === 1 ){
+  	return 1;
+  }
+  if( times === 1){
+  	return num;
+  }
+  if(times > 1) {
+   return num * power(num, times-1);
+  }
+}
+
+console.log(power(2, 4));
+
+//Reverse a string
+const reverse = str => {
+  return str.split("").reverse().join("");
+}
+
+console.log(reverse("Recursion Sucks"));
+
+const reverseRecur = str => {
+  var strNew = "";
+  for(let i=str.length-1;i>=0; i--){
+    strNew+=str.charAt(i);
```

Comment:
This is great. Think about what value you are checking for on each iteration and try to create a recursive function that will check the same value as well.